prov/efa/src/efa_hmem.c

```diff
@@ -67,6 +67,7 @@ static int efa_domain_hmem_info_init_protocol_thresholds(struct efa_domain *efa_
 	case FI_HMEM_SYSTEM:
 		/* We have not yet tested runting with system memory */
 		info->runt_size = 0;
+		info->max_gdrcopy_from_device_size = EFA_DEFAULT_INTRA_MAX_GDRCOPY_FROM_DEV_SIZE;
```

Comment:
The name is not quite right. Actually we not only copy from device, we also gdrcopy from host to device in the recv side.

Reply:
Yes this is on purpose. This value should only be used for device to host copy. gdrcopy claims asymmetrical performance between D2H and H2D. Here we are concerned with D2H.

Reply:
But finally you still use gdrcopy on receive side. You silently ignore its impact.. And your change on the protocol selection seems not explicitly force gdrcopy to be used, i.e. if gdrcopy is not available, it will still use cudamemcpy for msg smaller than the threshold. Is that true?

Reply:
The choice between gdrcopy and cudamemcpy always happens within prov/efa with this change.

>  if gdrcopy is not available, it will still use cudamemcpy for msg smaller than the threshold. Is that true
This is a good point. Let me think about it.

Reply:
@shijin-aws  You are right. The previous logic will cause regression if gdrcopy is not available - we would do 2 cudaMemcpy, which would double the latency.

I have updated the change to separate the threshold selection logic for FI_HMEM_SYSTEM and FI_HMEM_CUDA - and check if gdrcopy is enabled or not.